OpenRA.Mods.Common/Traits/SupportPowers/NukePower.cs

```diff
@@ -197,5 +218,73 @@ public void Activate(Actor self, WPos targetPosition)
 				});
 			}
 		}
+
+		public override void SelectTarget(Actor self, string order, SupportPowerManager manager)
+		{
+			Game.Sound.PlayToPlayer(SoundType.UI, manager.Self.Owner, Info.SelectTargetSound);
+			self.World.OrderGenerator = new SelectNukePowerTarget(self, order, manager, info, MouseButton.Left);
+		}
+	}
+
+	public class SelectNukePowerTarget : OrderGenerator
+	{
+		readonly Actor self;
+		readonly SupportPowerManager manager;
+		readonly NukePowerInfo info;
+		readonly string order;
+		readonly MouseButton expectedButton;
+
+		public SelectNukePowerTarget(Actor self, string order, SupportPowerManager manager, NukePowerInfo info, MouseButton button)
+		{
+			// Clear selection if using Left-Click Orders
+			if (Game.Settings.Game.UseClassicMouseStyle)
+				manager.Self.World.Selection.Clear();
+
+			this.self = self;
+			this.manager = manager;
+			this.info = info;
+			this.order = order;
+			expectedButton = button;
+		}
+
+		protected override IEnumerable<Order> OrderInner(World world, CPos cell, int2 worldPixel, MouseInput mi)
+		{
+			world.CancelInputMode();
+			if (mi.Button == expectedButton && world.Map.Contains(cell))
+				yield return new Order(order, manager.Self, Target.FromCell(world, cell), false) { SuppressVisualFeedback = true };
+		}
+
+		protected override void Tick(World world)
+		{
+			// Cancel the OG if we can't use the power
+			if (!manager.Powers.ContainsKey(order))
+				world.CancelInputMode();
+		}
+
+		protected override IEnumerable<IRenderable> Render(WorldRenderer wr, World world) { yield break; }
+		protected override IEnumerable<IRenderable> RenderAboveShroud(WorldRenderer wr, World world) { yield break; }
+
+		protected override IEnumerable<IRenderable> RenderAnnotations(WorldRenderer wr, World world)
+		{
+			if (!info.EnableRangeCircles)
+				yield break;
+
+			var warheads = info.WeaponInfo.Warheads.OfType<SpreadDamageWarhead>();
+			var centerPosition = wr.World.Map.CenterOfCell(wr.Viewport.ViewToWorld(Viewport.LastMousePos));
+			foreach (var warhead in warheads)
+				yield return new RangeCircleAnnotationRenderable(
+					centerPosition,
+					warhead.Spread,
```

Comment:
This doesn't sound right. The total spread area is affected by length of Falloff too and the total spread can be defined with Range too, which overrides Spread.

Just using Spread without Falloff is gonna give the area that's affected under first Falloff value, which is by default the full damage. But taking Falloff into action will also look wrong with nuke warheads in base mods, as last Fallout circle only deals 0.7% of full damage. Not that the default 5% is high either.

Which makes me question if it is a good idea to calculate this automatically. Even if it is calculated automatically with or without taking Falloff into action, i think it would be good to be able to override this in the Power itself, with an WDist array (so multipile can be defined).

Reply:
I tried using range, but that gave way to large radii. While indeed there will be tree fires in that radius, it is insignificant gameplay wise. Spread is a good compromise as it mimics the Red Alert 2 visuals while also being based on real weapon stats:

![TD example](https://media.discordapp.net/attachments/153649279762694144/698238104531894424/unknown.png)

Reply:
@MustaphaTR is right - the current setup is hardcoding for the way that RA and TD have defined their yaml, but will fail in most other cases. This doesn't even work for D2k or TS!

Reply:
I see. Went with something that is fully custom configurable now:

![image](https://user-images.githubusercontent.com/756669/79041048-fa4bee80-7bec-11ea-86d4-f9ed84f786b4.png)